lib/auth/webauthn/messages.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webauthn
+
+import "github.com/duo-labs/webauthn/protocol"
+
+// CredentialAssertion is the payload sent to authenticators to initiate login.
+type CredentialAssertion protocol.CredentialAssertion
+
+// CredentialAssertionResponse is the reply from authenticators to complete
+// login.
+type CredentialAssertionResponse struct {
+	// CredentialAssertionResponse is redefined because, unlike
+	// CredentialAssertion, it is likely to be manually created by package users.
+	// Redefining allows us to 1) make sure it can be properly JSON-marshaled
+	// (protocol.CredentialAssertionResponse.Extensions can't) and 2) we avoid
+	// leaking the duo-labs/webauthn dependency.
+	// The nesting of types is identical to protocol.CredentialAssertionResponse.
+
+	PublicKeyCredential
+	AssertionResponse AuthenticatorAssertionResponse `json:"response"`
+}
+
+type PublicKeyCredential struct {
+	Credential
+	RawID      protocol.URLEncodedBase64              `json:"rawId"`
```

Comment:
Does "raw" mean still base64 encoded?

Reply:
Just double-checked, it is correct. The standard defines [rawId as an ArrayBuffer](https://www.w3.org/TR/webauthn-2/#iface-pkcredential), but I suppose one must encode it anyway for transport.